http/http.go

```diff
@@ -23,11 +23,22 @@ type Client interface {
 	Do(req *http.Request) (*http.Response, error)
 }
 
+var _ http.ResponseWriter = &responseWriterObserver{}
+var _ http.Flusher = &responseWriterObserver{}
+
 type responseWriterObserver struct {
 	w  http.ResponseWriter
 	sc int
 }
 
+func (w *responseWriterObserver) Flush() {
```

Comment:
shoot, this is a bummer. this means downstream users can no longer use type assertion to know if calling Flush is safe or not.

what do you think about taking the approach here? https://pkg.go.dev/gopkg.in/webhelp.v1@v1.0.0-20170530084242-3f30213e4c49/whmon#MonitorResponse

basically, the type returned implements Flusher only if the internal type also implements Flusher

Reply:
maybe the more important method is wrapResponseWriter here: https://github.com/go-webhelp/webhelp/blob/3f30213e4c49/whmon/writer.go#L85

Reply:
That's a great trick, thanks for the idea.